app/Services/Pocket.php

```diff
@@ -0,0 +1,221 @@
+            ->limit(1)
+            ->get()
+            ->first();
+
+        if ($userItem == null) {
+            $userItem = new UserItemsModel();
+            $userItem->user_id = $this->user->id;
+            $userItem->item_id = $item->id;
+            $userItem->save();
+        }
+    }
+
+    protected function addItem($url, $title = null, $excerpt = null)
+    {
+        if (ItemModel::find(md5($url))) {
+            return ItemModel::find(md5($url));
+        }
+
+        if (!$articleText = $this->getArticleContent($url)) {
+            Log::error('Unable to find article for URL: ' . $url);
+        }
+
+        $item = new ItemModel();
+        $item->id = md5($url);
+        $item->url = $url;
+        $item->content = $this->replace4byte($articleText);
+        $item->title = $this->replace4byte($title);
+        $item->excerpt = $this->replace4byte(substr($excerpt, 0, 255));
+        $item->status = ItemModel::STATUS_FETCHED;
+        $item->save();
+
+        return $item;
+    }
+
+    /**
+     * Get article body text
+     * First try Goose, then try Readability
+     *
+     * @param $url
+     * @return string
+     */
+    protected function getArticleContent($url)
+    {
+        $goose = new GooseClient();
+
+        try {
+            $article = $goose->extractContent($url);
+            $articleText = $article->getCleanedArticleText();
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+            return false;
+        }
+
+        if (!$articleText) {
+            // If Goose failed, try Readability
+            $html = $article->getRawHtml();
+
+            if (function_exists('tidy_parse_string')) {
+                $tidy = tidy_parse_string($html, array(), 'UTF8');
+                $tidy->cleanRepair();
+                $html = $tidy->value;
+            }
+
+            $readability = new Readability($html, $url);
+            $readability->debug = false;
+            $result = $readability->init();
+
+            if ($result) {
+                $articleText = $readability->getContent()->innerHTML;
+                // if we've got Tidy, let's clean it up for output
+                if (function_exists('tidy_parse_string')) {
+                    $tidy = tidy_parse_string($articleText, array('indent' => true, 'show-body-only' => true), 'UTF8');
+                    $tidy->cleanRepair();
+                    $articleText = $tidy->value;
+                }
+            }
+        }
+
+        $articleText = trim(strip_tags($articleText));
+
+        return $articleText;
+    }
+
+    /**
+     * Replace utf8mb4 with utf8
+     * @param $string
+     *
+     * @return mixed
+     */
+    protected function replace4byte($string) {
+        return preg_replace('%(?:
+          \xF0[\x90-\xBF][\x80-\xBF]{2}      # planes 1-3
+        | [\xF1-\xF3][\x80-\xBF]{3}          # planes 4-15
+        | \xF4[\x80-\x8F][\x80-\xBF]{2}      # plane 16
+        )%xs', '', $string);
```

Comment:
emoticons are in utf8mb4 which isn't supported in the out of the box mysql so I have to strip any characters that aren't UTF8.